src/h5web/dataset-visualizer/VisProvider.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { createContext, ReactElement, ReactNode } from 'react';
+import { DimensionMapping } from './models';
+import { HDF5Dataset } from '../providers/models';
+import { useValue } from '../providers/hooks';
+import { isSimpleShape } from '../providers/utils';
+
+export interface VisProps {
+  mapping: DimensionMapping;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  data: any;
+  dims: number[];
+}
+
+export const VisContext = createContext<VisProps | undefined>(undefined);
+
+type Props = {
+  key: string;
+  mapping: DimensionMapping;
+  dataset: HDF5Dataset;
+  children: ReactNode;
+};
+
+export function VisProviderForCanvas(
+  props: VisProps & { children: ReactNode }
+): ReactElement {
+  const { mapping, dims, data, children } = props;
+
+  return (
+    <VisContext.Provider value={{ data, dims, mapping }}>
+      {children}
+    </VisContext.Provider>
+  );
+}
```

Comment:
R3F's `Canvas` require that we provide the context once again. To avoid triggering `useValue` once more and problems due to the mismatch of input/output props, we need a special version of the provider that only forwards the props.

Reply:
Looking at the code, I don't think you need a component for this. Just use `<VisContext.Provider {...providerProps}>` in the visualizations directly.